app/models/miq_expression.rb

```diff
@@ -575,7 +575,7 @@ def self._to_ruby(exp, context_type, tz)
   def to_sql(tz = nil)
     tz ||= "UTC"
     @pexp, attrs = preprocess_for_sql(@exp.deep_clone)
-    sql = to_arel(@pexp, tz).to_sql if @pexp.present?
+    sql = Expression.build(@pexp, tz).to_sql if @pexp.present?
```

Comment:
> Is it possible and does it makes sense to have something that looks like this (in terms of the timezone):

> 
```ruby
if @pexp.present?
  exp_obj = Expression.build(@pexp.to_sql)
  exp_obj.timezone = tz if exp_obj.timezone_required?
  sql = exp_obj.to_sql
end
```

Have given this some thought. I love that this makes this hidden concept (i.e. why/when is a timezone required?) explicit. What makes me a little uneasy about this though is that it means we're creating objects that might not have everything they need on construction - components that need timezones are only good to go after they have been optionally set. I also think it doesn't address the fact that all these objects will still have `timezone` data, which it seems to me is the core of the cohesion problem.

Ideally I'd like to separate the timezone stuff out so that components wouldn't have to know anything about them, they'd just have operators and operands, and know what to do with them. But I don't think it can currently be done - a concept like "today" is dependent on the operator and the column type.

I guess it could be better to do the datetime normalization at the building stage (so that instances would know nothing about timezones)? I realize though that that doesn't address your concern. If you feel strongly about it I am happy to implement something similar to what you suggested!